### DISCUSSION TITLE
Automatically add session cookie id to requests after logged in

### DISCUSSION THREAD

**Discussion by sidalit** [maxcountryman/axum-login#231] [Q&A] [answered]
Hello

I would like to understand how to automatically add cookies to each protected requests.

I am building a backend that should handle an API (only if user is logged in).
You can find a very simplified example in [this repo](https://github.com/syedelec/axum-temp) that can be run with `cargo run`

Here is a screenshot of the output:

![image](https://github.com/maxcountryman/axum-login/assets/9877335/8111aafb-13b1-41f9-9cf5-7b3222cfa55f)

First I do the POST /login, then the GET /api/secret and then GET /api/secret (with cookie id copied/pasted manually into postman/bruno)

How can I automatically handle logged in user and automatically have cookie used for requests instead of adding it manually.

Thanks!

**Comment by maxcountryman** [comment]
> I would like to understand how to automatically add cookies to each protected requests.

I'm not really sure what you mean by this. Cookies are generally added to requests by a user agent (such as a web browser which will include the cookie if it's appropriate to do so). If you're using some other client, you may need to include cookies in the request manually (such as with cURL).

**Reply by sidalit** [reply]
They should be handled automatically with the client API I am using : postman and bruno
At least, when I try the `axum-login` examples using these tools, it works well.

**Reply by maxcountryman** [reply]
I can't speak for Postman, but if it's able to include the cookie automatically then it should do so; yes.

**Reply by sidalit** [reply, marked as answer]
The issue was due to `with_secure(true)`, I changed it to `with_secure(false)` and everything is working as expected.
Thanks!

**Reply by maxcountryman** [reply]
While that's fine for local development and testing, you almost certainly want to set this to `true` in production

**Reply by sidalit** [reply]
Right, sure :+1:
